lib/kv-extreme-scale.js

```diff
@@ -0,0 +1,370 @@
+'use strict';
+
+var assert = require('assert');
+var connectorCore = require('loopback-connector');
+var debug = require('debug')('loopback:connector:kv-extreme-scale');
+var extend = require('util')._extend;
+var request = require('request');
+var util = require('util');
+
+var Connector = connectorCore.Connector;
+var BinaryPacker = connectorCore.BinaryPacker;
+var ModelKeyComposer = connectorCore.ModelKeyComposer;
+
+exports.initialize = function initializeDataSource(dataSource, callback) {
+  var settings = dataSource.settings;
+
+  dataSource.connector = new ExtremeScaleKVConnector(settings, dataSource);
+
+  if (!callback) return;
+  return process.nextTick(callback);
+};
+
+exports._Connector = ExtremeScaleKVConnector;
+
+/**
+ * @constructor
+ *
+ * Constructor for the KeyValue ExtremeScale connector.
+ *
+ * @param {Object} settings
+ * @param {DataSource} dataSource The data source instance.
+ *
+ * @header ExtremeScaleKeyValueConnector(settings, dataSource)
+ */
+function ExtremeScaleKVConnector(settings, dataSource) {
+  Connector.call(this, 'kv-extreme-scale', settings);
+  this.dataSource = dataSource;
+
+  debug('Connector settings', settings);
+
+  // set key expiration based on LastUpdateTime with Optimistic locking
+  this._url = settings.url + '.LUT.O';
+
+  this._queryUrl = this._url
+    .replace(/\/v1\/grids\//, '/v1/query/')
+    .replace(/\/$/, '');
+
+  this._strictSSL = settings.strictSSL;
+  this._cookieJar = request.jar();
+  this._packer = new BinaryPacker();
+  this.DataAccessObject = dataSource.juggler.KeyValueAccessObject;
+};
+
```

Comment:
I prefer to add an empty line after each method. In my view, the `inherits` line is a standalone statement that should stand out.